File: 2016/Ordenacao/SelectionSort.c
// Programa p25.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define INFINITO 999999
#define TAM_MAX  50

void SelectionSort(int x[], int n);
void troca(int v[], int i, int j);

void troca(int v[], int i, int j){   //funcao para trocar dois elementos
  int aux;
  aux = v[i];
  v[i] = v[j];
  v[j] = aux;
}

void SelectionSort(int x[], int n){
  int menor,pos;
  int i,k = 0;

  while (k < n){
    menor = INFINITO;
    for (i = k; i < n; i++)
      if (x[i] < menor){
        menor = x[i];
        pos = i;
      }
    troca(x,k,pos);
    k++;
  }
}

int main(){
  int i,n,a[TAM_MAX];

  // Inicializar gerador de números aleatórios
  srand((unsigned)time(NULL));

  // Vetor gerado aleatoriamente
  n = (rand() % TAM_MAX)+1;
  printf("Vetor original: ");
  for (i = 0; i < n; i++)
  {
    a[i] = (rand() % 100);
    printf("%d ",a[i]);
  }
  printf("\n");

  // Classificar vetor
  SelectionSort(a,n);
  printf("Vetor ordenado: ");
  for (i = 0; i < n; i++)
    printf("%d ",a[i]);
  printf("\n");
  system("PAUSE");
  return 0;
}

File: UNP/ref/Java/#ED2-Java-BKP/ordenacao_busca/SelectionSort.java
package ordenacao_busca;

public class SelectionSort {

    public static void main(String[] args) {
        int[] array_desordenado1 = {9,8,7,6,5,4,3,2,1};
        int[] array_desordenado2 = {1,5,9,7,8,8,10,20,2,-1};

        int[] array_ordenado = selectionsort(array_desordenado1);
        int[] array_ordenado2 = selectionsort(array_desordenado2);

        System.out.println("Exibindo o array 1 ordenado: ");
        for (int i: array_ordenado) {
            System.out.print(i + " ");
        }

        System.out.println("\n\nExibindo o array 2 ordenado: ");
        for (int i: array_ordenado2) {
            System.out.print(i + " ");
        }
    }

    private static int[] selectionsort(int[] vetor) {
        int n = vetor.length;
        for (int i = 0; i < n; i++) {
            int menor = i;
            for (int j = i+1; j < n; j++) {
                if(vetor[j]<vetor[menor])
                    menor=j;
            }
            //menor encontrado, basta trocar
            troca(vetor, i, menor);
        }
        return vetor;
    }

    private static void troca(int[] lista, int i, int j) {
        int aux = lista[i];
        lista[i] = lista[j];
        lista[j] = aux;
    }
}
